File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    // Moores Voting Algorithm TC:O(n) SC:O(1)
    int majorityElement(vector<int>& nums) {
        
        int count=0 , ele=0;
        
        for(int i=0 ;i<nums.size();i++)
        {
            if(count==0)
                ele=nums[i];
            
            if(ele==nums[i])
                count++;
            
            else
                count--;
        }
        return ele;
    }
};

File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        
        HashMap<Integer,Integer>m=new HashMap<>();
        
        for(int i=0 ;i<nums.length ;i++)
        {
            if(!m.containsKey(nums[i]))
            {
                m.put(nums[i],1);
            }   
            else 
            {  
            
                m.put(nums[i],m.get(nums[i])+1);
            }
        }
        
         
        for (Map.Entry<Integer,Integer> e : m.entrySet())
        {
            if(e.getValue()>nums.length/2)
            {
                return e.getKey();
            }
        }
        
       return -1;
    }
}